serde sympy functions

avikchaudhuri opened the pull request:
Summary: Sympy functions appearing in symbolic expressions inside tensor metadata were not being deserialized properly.

Test Plan: updated test

Differential Revision: D60573150

pianpwk commented on torch/_export/serde/serialize.py:
```diff
@@ -1850,6 +1853,11 @@ def deserialize(
+                "ToFloat": torch.utils._sympy.functions.ToFloat,
+                "TruncToInt": torch.utils._sympy.functions.TruncToInt,
+                # add more sympy functions here
+            }
```
question, not comment: we need to add all functions here right? Not just the unsupported ones

avikchaudhuri replied:
Yeah, but I don't know which ones should be here vs. handled elsewhere; some of them are int operations, like Mod and FloorDiv, should they be here? Or is it just the floor <-> int ones?

If you're going to do a pass anyway, maybe you can fill this up? Or let me know what else to add here.

avikchaudhuri replied:
Added all.

avikchaudhuri committed:
serde sympy functions (#132493)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/132493

Sympy functions appearing in symbolic expressions inside tensor metadata were not being deserialized properly.

Test Plan: updated test

Differential Revision: D60573150

avikchaudhuri commented:
@pytorchbot merge -f "unrelated failure"
